### PR TITLE
Luiz/Submenu feature

### PR DESCRIPTION
**Pull request by luiztiago**
## Notes
* To use the parent item, should set the $menuParent on class extending `AbstractController`

### CODE REVIEW

File: Menu/MenuBuilder.php [hunk 1]
```diff
@@ -34,7 +34,25 @@ public function createAdminMenu(array $options)
...
         foreach ($this->controllerChain->getMenuAwareControllers() as $menuAwareController) {
             foreach ($menuAwareController->getMenuRoutes() as $menuRoute) {
-                $menu->addChild($menuRoute['title'], ['route' => $menuRoute['route'], 'linkAttributes' => ['icon' => $menuAwareController->getMenuIcon()]]);
+                $route = ['route' => $menuRoute['route'], 'linkAttributes' => ['icon' => $menuAwareController->getMenuIcon()]];
+
+                if (method_exists($menuAwareController, 'getMenuParent')) {
+                    $menuParent = $menuAwareController->getMenuParent();
+
+                    if (is_null($menuParent)) {
+                        continue;
```
Comment (johnmadrak) [inline Menu/MenuBuilder.php]: @luiztiago - If ```$menuParent``` is null here you will not add the navigation option. Is this intentional?

File: Controller/AbstractController.php [hunk 2]
```diff
@@ -92,6 +93,14 @@ public function getRelatedRoutes()
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function getMenuParent()
```
Comment (johnmadrak) [inline Controller/AbstractController.php]: @luiztiago - What do you think of "Group" instead of Parent?

File: Menu/MenuBuilder.php (hunk 1 above)
Comment (luiztiago) [inline Menu/MenuBuilder.php]: @johnmadrak - Good catch! Fixed! :)

File: Controller/AbstractController.php (hunk 2 above)
Comment (luiztiago) [inline Controller/AbstractController.php]: @johnmadrak - Good idea! renamed :)

File: Menu/MenuBuilder.php [hunk 3]
```diff
@@ -34,7 +34,23 @@ public function createAdminMenu(array $options)
 
         foreach ($this->controllerChain->getMenuAwareControllers() as $menuAwareController) {
             foreach ($menuAwareController->getMenuRoutes() as $menuRoute) {
-                $menu->addChild($menuRoute['title'], ['route' => $menuRoute['route'], 'linkAttributes' => ['icon' => $menuAwareController->getMenuIcon()]]);
+                $route = ['route' => $menuRoute['route'], 'linkAttributes' => ['icon' => $menuAwareController->getMenuIcon()]];
+
+                if (method_exists($menuAwareController, 'getMenuGroup')) {
+                    $menuGroup = $menuAwareController->getMenuGroup();
+
+                    if (is_null($menuGroup) === false) {
```
Comment (johnmadrak) [inline Menu/MenuBuilder.php]: @luiztiago - Could we do something like:

```if (method_exists($menuAwareController, 'getMenuGroup') === true && empty($menuAwareController->getMenuGroup()) === false) {``` 
on line 39?

Comment (luiztiago) [inline Menu/MenuBuilder.php]: @johnmadrak - Good idea! Fixed

**Comment by johnmadrak** [comment]
👍

**Pull request merged**